### ISSUE TITLE
Allow displaying non-Recommended add-ons in Primary Hero area

### ISSUE THREAD

**Issue by devaneymoz** [mozilla/addons-server#15671]
In AMO admin under Promoted Addons, if an extension's designated Group is Recommended it can be featured in the Hero area of the homepage. We should consider expanding Hero access to the following other Group designations: 

- Sponsored
- By Firefox
- Spotlight

@jvillalobos

**Comment by jvillalobos** [comment]
Here's what I think would be needed for a first iteration of this:

- Allow in the admin to add any promoted add-on to the primary hero area
- The header should say: `Recommended` for Recommended add-ons, `By Firefox` for By Firefox add-ons, `Sponsored` for anything else. If `Sponsored`, the "What is this?" link should appear next to it (design tbd).
- We'll need new GA events for impressions and clicks, also tbd.
- No Adzerk integration at this point.

I'm thinking we may be able to squeeze this into the Q4 plan so we can use it for testing in Q1, but we still need to discuss how this will fit into the roadmap and get the missing bits (mainly the link design).

**Comment by eviljeff** [comment]
> - The header should say: Recommended for Recommended add-ons, By Firefox for By Firefox add-ons, Sponsored for anything else. If Sponsored, the "What is this?" link should appear next to it (design tbd).
> - We'll need new GA events for impressions and clicks, also tbd.

Can someone split this work out into an addons-frontend issue?  I'd recommend that work be done to take the promoted group into account before any changes to the admin on addons-server

**Comment by eviljeff** [comment]
admin changes are complete and will be live on October 22nd though I suggest you avoid adding any non-recommended addons to primary hero items until the frontend issues are also closed.

**Comment by bobsilverberg** [comment]
I didn't realize you were going to get to this so quickly, but I can try to make the related frontend changes to land on Oct. 22 as well.

**Comment by ioanarusiczki** [comment]
@eviljeff 
I verified it on dev - FF81(Win10). I think the promoted model works as requested meaning:

- recommended, sponsored, by Firefox and Spotlight addons can be added to the primary hero shelf from [admin](https://addons-internal.dev.mozaws.net/en-US/admin/models/discovery/promotedaddon/).
- not recommended, verified or strategic addons cannot be added. The message is " Only add-ons that are Recommended, Sponsored, By Firefox, Spotlight can be enabled for non-external primary shelves."

One thing I've noticed is that the  module allows adding promoted addons on the primary hero shelf from all 3 Application types (all, firefox or android).
So if I'm adding a recommended addon to the primary hero shelf which has "Android" application set -> the homepage will display it on desktop. The android platform does not display the primary hero section anyway.

Let me know if I should file an issue or it's a known limitation.
